<launch>
  <arg name="voxel_size" default="0.10"/>
  <arg name="generate_esdf" default="false" />

   <node name="voxblox_node" pkg="voxblox_ros" type="voxblox_node" output="screen" args="-alsologtostderr" clear_params="true">
    <param name="tsdf_voxel_size" value="$(arg voxel_size)" />
    <param name="tsdf_voxels_per_side" value="16" />
    <param name="voxel_carving_enabled" value="true" />
    <param name="color_mode" value="color" />
    <param name="use_tf_transforms" value="false" />
    <param name="update_mesh_every_n_sec" value="1.0" />
    <param name="min_time_between_msgs_sec" value="0.2" />
    <param name="method" value="merged" />
    <param name="generate_esdf" value="$(arg generate_esdf)" />
    <param name="use_const_weight" value="false" />
    <param name="allow_clear" value="true" />
    <!--<param name="truncation_distance" value="0.5" /> -->
    <param name="verbose" value="true" />
    <param name="mesh_filename" value="$(find voxblox_ros)/mesh_results/$(anon tsdf_convert).ply" />
  </node>

</launch>
